src/Chapter3.hs

```diff
@@ -1006,6 +1150,45 @@ Implement data types and typeclasses, describing such a battle between two
 contestants, and write a function that decides the outcome of a fight!
 -}
 
+type Hlt = Int
+type Atk = Int
+type Def = Int 
+-- Used the names beast and hero because I'm pretty
+-- sure Knight and monster were used in previous tasks 
+data Hero = Hero 
+  { heroHealth :: Hlt
+  , heroDefence :: Def
+  , heroAttack :: Atk
+  }
+
+castSpell :: Hero -> Hero
+castSpell hero = hero {heroDefence = heroDefence hero + 1}
+
+drinkPotion :: Hero -> Hero
+drinkPotion hero = hero {heroHealth = heroHealth hero + 1}
+
+data Beast = Beast
+  { beastHealth :: Hlt
+  , beastAttack :: Atk
+  }
+
+flee :: Beast -> Beast
+flee beast = beast {beastHealth = 0} 
+
+class Attack target where
+  attack :: Either Hero Beast -> target -> target 
```

Comment:
Instead of `Either Hero Beast` it is better to use the polymorphic value, which should be of the class `Fighter`